apps/teams/tests/test_member_management.py

```diff
@@ -134,9 +139,13 @@ def test_only_admin_cant_leave(self):
         c.force_login(self.admin)
 
         # demote other admin
-        self._change_role(c, self.admin_membership2, roles.ROLE_MEMBER)
+        self._change_role(c, self.admin_membership2, self.member_groups)
 
         # confirm it doesn't work
         response = self._remove_member(c, self.admin_membership)
         self.assertNotEqual(200, response.status_code)
         self.assertTrue(Membership.objects.filter(pk=self.admin_membership.pk).exists())
+
+    def _check_groups(self, membership, expected_group_ids):
+        membership = Membership.objects.get(id=membership.id)
```

Comment:
just out of curiosity, no need to really change anything: is there a reason why you don't just use the membership object from the params directly instead of doing `Membership.objects.get(id=membership.id)`?

Reply:
I need to reload it from the DB to clear the M2M cache.

I'll add a comment.